src/stream/src/common/log_store_impl/kv_log_store/reader.rs

```diff
@@ -192,6 +192,10 @@ pub struct KvLogStoreReader<S: StateStoreRead> {
     rewind_delay: RewindDelay,
 
     align_epoch_on_init: bool,
+
+    rewind_start_offset: Option<u64>,
+
+    aligned_range_start: Option<LogStoreReadStateStreamRangeStart>,
```

Comment:
We can make `aligned_range_start` the as a field of a new variant named like `KvLogStoreReaderFutureState::Reset`, since it's always used in `build_stream_from_start_offset`. In `build_stream_from_start_offset` we can assert that we must be at `KvLogStoreReaderFutureState::Reset`.